### PR TITLE
ITS and MFT dead damp builder workflow

### PR DESCRIPTION
**Pull request by nicolovalle**
This PR replaces https://github.com/AliceO2Group/AliceO2/pull/12442 and https://github.com/AliceO2Group/AliceO2/pull/12445

cc @shahor02 @mcoquet642 , @iravasen @IsakovAD

### CODE REVIEW

**Comment by iravasen** [comment]
Thanks a lot @nicolovalle for the very fast implementation! It looks fine to me since you set also the process name based on the detector.

File: Detectors/ITSMFT/common/workflow/src/DeadMapBuilderSpec.cxx [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+    const auto ROFs = pc.inputs().get<gsl::span<o2::itsmft::ROFRecord>>("ROFs");
+    for (const auto& rof : ROFs) {
+      auto elementsInTF = rof.getROFData(elements);
+      for (const auto& el : elementsInTF) {
+        uint16_t chipID = (uint16_t)el.getSensorID();
+        mDeadElements.erase(getElementIDFromChip(chipID));
+      }
+    }
+  } else if (mDataSource == "chipsstatus") {
+    const auto elements = pc.inputs().get<std::vector<char>>("elements");
```
Comment (shahor02) [inline Detectors/ITSMFT/common/workflow/src/DeadMapBuilderSpec.cxx]: should be 
```
const auto elements = pc.inputs().get<gsl::span<char>>("elements");
```

**Review by shahor02 — COMMENTED** [review]
Hi @nicolovalle 

Please see 1 comment below (vectors of fixed size objects should be accessed as spans).
The rest looks fine assuming you tested the workflow.
One general comment is that inserting/erasing in std::set is a relatively expensive procedure, would be faster to use a simple vector. But given that you are processing 1 TF out of many this should also work

File: Detectors/ITSMFT/common/workflow/src/DeadMapBuilderSpec.cxx (hunk 1 above)
Comment (nicolovalle) [reply Detectors/ITSMFT/common/workflow/src/DeadMapBuilderSpec.cxx]: Thanks Ruben, optimized in the new commit. 
The code has been tested for ITS and it produces meaningful maps.

**Review by shahor02 — APPROVED** [review]
Thanks! Will merge once CI is passed.

**Pull request merged**